File: frontend/src/modules/brands/types/brand.ts
export type MarcaEstado = 'activo' | 'inactivo'

export interface Marca {
  id: number
  nombre: string
  created_at: string
  updated_at: string
}

export interface MarcaFilter {
  q?: string
}

export interface MarcaCreate {
  nombre: string
}

export interface MarcaUpdate {
  nombre: string
}

File: backend/app/schemas/brand.py
from datetime import datetime
from pydantic import BaseModel, ConfigDict


class MarcaCreate(BaseModel):
    nombre: str


class MarcaUpdate(BaseModel):
    nombre: str


class MarcaResponse(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: int
    nombre: str
    created_at: datetime
    updated_at: datetime
